middleware/authorization.go

```diff
@@ -1,25 +1,22 @@
 package middleware
 
 import (
-	"context"
 	"fmt"
 	"net/http"
-	"os"
 	"regexp"
-	"time"
 
-	"github.com/RedHatInsights/rbac-client-go"
 	"github.com/RedHatInsights/sources-api-go/config"
 	h "github.com/RedHatInsights/sources-api-go/middleware/headers"
+	"github.com/RedHatInsights/sources-api-go/rbac"
 	"github.com/RedHatInsights/sources-api-go/util"
 	"github.com/labstack/echo/v4"
 	"github.com/redhatinsights/platform-go-middlewares/identity"
 )
 
 var (
-	psks            = config.Get().Psks
-	bypassRbac      = config.Get().BypassRbac
-	rbacClient Rbac = &RbacClient{client: rbac.NewClient(os.Getenv("RBAC_URL"), "sources")}
+	psks       = config.Get().Psks
+	bypassRbac = config.Get().BypassRbac
+	rbacClient = rbac.NewRbacClient(config.Get().RbacHost)
```

Comment:
It would be nice to explicitly inject these into the `PermissionCheck()` method and remove the globals.  This would likely make testing easier.

It doesn't have to be done now...just something to think about.

Reply:
Yes, wholeheartedly agree! For Kessel's refactor I was telling Dalia to pass the Kessel dependencies like you suggested, but I was unsure if I should do it in this PR too in case it'd be out of scope. However your comment has been a great excuse to do it!

Refactored!